module-4/backend/week_two.md

```diff
@@ -1,22 +1,35 @@
+* Asynchronous JavaScript takes advantage of the browsers ability to run more than one function at a time. It would first fire off a method that would come on to the stack and then be sent to a Web API so that it is no longer on the stack and be able to continue to execute code/functions before the response of the first function is complete. This can be advantageous, but should be used in a limited way otherwise the benefits are lost.
 3. What are the four pillars of Object Oriented programming?
+* Abstraction: The idea of pulling out parts of code for reusability and to follow the DRY and SRP.
+* Encapsulation: The idea of grouping functions together base on the shared similar behavior and attributes.
+* Inheritance: The idea of passing on functionality to another class that can use and build upon the functionality provided by that class.
+* Polymorphism: The idea that a function can be used in a different way to execute similar behavior. For example, a Model T has an engine and a Corvette has an engine. Though both have engines there are parts of the engine that are different though the concept of the engine is similar.
 4. What are some tools available in JavaScript to help you write object oriented code?
+* Classes are a tool provided by JavaScript to help write more object oriented code.
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+* Being aware of repetition, functions that are too long, naming of variables and functions that are not clear, leaving in debuggers statements are all things to look for in code.
 6. What's a callback function and what are some reasons when we use/need callback functions?
+* Call back functions are functions that are passed to another function as a parameter. These are good to use in promises.
 7. What's the scope of variables in Javascript?
+* The scope of variable are dependent upon how the variable is declared.
+For instance, let is scoped only to the block within which it is defined. Const variables are defined once and cannot be later redefined in the code. Var is usually scoped to where it is defined. If defined as a global variable it is available anywhere throughout the code, where as if it is defined within the function it is only available within that function.
 8. What's the difference between `==` and `===` in JavaScript?
-9. Why do front end frameworks exist?
-
+* `==` is considered a loose comparison where `4 == '4'` would evaluate to true though the types are different it will coerce them into the same type. Where as the `===` is a strict comparison operator and `4 === '4'` will return false because it compares the type.
+9. Why do front-end frameworks exist?
+* Front-end frameworks exist to help provide structure and some organization for files and code. It helps put in place rules to follow for better ease of navigation.
```

Comment:
Yes! It also bundles up basic functionality that makes our lives easier as developers.